src/transcript.rs

```diff
@@ -284,3 +284,25 @@ impl<C: CurveAffine, Type> Deref for ChallengeScalar<C, Type> {
         &self.inner
     }
 }
+
+pub(crate) fn read_n_points<C: CurveAffine, R: Read, T: TranscriptRead<R, C>>(
```

Comment:
I think this should be a `read_points` method on `TranscriptRead` (or perhaps a new `TranscriptReadExt`) with a default implementation (and likewise for `read_scalars`).